src/measure_theory/interval_integral.lean

```diff
@@ -0,0 +1,274 @@
+
+lemma integral_of_le (h : a ≤ b) : ∫ x in a..b, f x ∂μ = ∫ x in Ioc a b, f x ∂μ :=
+by simp [interval_integral, h]
+
+@[simp] lemma integral_same : ∫ x in a..a, f x ∂μ = 0 :=
+sub_self _
+
+lemma integral_symm (a b) : ∫ x in b..a, f x ∂μ = -∫ x in a..b, f x ∂μ :=
+by simp only [interval_integral, neg_sub]
+
+lemma integral_of_ge (h : b ≤ a) : ∫ x in a..b, f x ∂μ = -∫ x in Ioc b a, f x ∂μ :=
+by simp only [integral_symm b, integral_of_le h]
+
+lemma integral_cases (f : α → E) (a b) :
+  ∫ x in a..b, f x ∂μ ∈ ({∫ x in Ioc (min a b) (max a b), f x ∂μ,
+    -∫ x in Ioc (min a b) (max a b), f x ∂μ} : set E) :=
+(le_total a b).imp (λ h, by simp [h, integral_of_le]) (λ h, by simp [h, integral_of_ge])
+
+lemma norm_integral_eq_norm_integral_Ioc :
+  ∥∫ x in a..b, f x ∂μ∥ = ∥∫ x in Ioc (min a b) (max a b), f x ∂μ∥ :=
+(integral_cases f a b).elim (congr_arg _) (λ h, (congr_arg _ h).trans (norm_neg _))
+
+lemma norm_integral_le_integral_norm_Ioc :
+  ∥∫ x in a..b, f x ∂μ∥ ≤ ∫ x in Ioc (min a b) (max a b), ∥f x∥ ∂μ :=
+calc ∥∫ x in a..b, f x ∂μ∥ = ∥∫ x in Ioc (min a b) (max a b), f x ∂μ∥ :
+  norm_integral_eq_norm_integral_Ioc
+... ≤ ∫ x in Ioc (min a b) (max a b), ∥f x∥ ∂μ :
+  norm_integral_le_integral_norm f
+
+lemma norm_integral_le_abs_integral_norm : ∥∫ x in a..b, f x ∂μ∥ ≤ abs (∫ x in a..b, ∥f x∥ ∂μ) :=
+begin
+  simp only [← real.norm_eq_abs, norm_integral_eq_norm_integral_Ioc],
+  exact le_trans (norm_integral_le_integral_norm _) (le_abs_self _)
+end
+
+lemma norm_integral_le_of_norm_le_const_ae {a b C : ℝ} {f : ℝ → E}
+  (h : ∀ᵐ x, x ∈ Ioc (min a b) (max a b) → ∥f x∥ ≤ C) :
+  ∥∫ x in a..b, f x∥ ≤ C * abs (b - a) :=
+begin
+  rw [norm_integral_eq_norm_integral_Ioc],
+  have : volume (Ioc (min a b) (max a b)) = ennreal.of_real (abs (b - a)),
+  { rw [real.volume_Ioc, max_sub_min_eq_abs, ennreal.of_real] },
+  rw [← ennreal.to_real_of_real (abs_nonneg _), ← this],
+  refine norm_set_integral_le_of_norm_le_const_ae'' _ is_measurable_Ioc h,
+  simp only [this, ennreal.lt_top_iff_ne_top],
+  exact ennreal.of_real_ne_top
+end
+
+lemma norm_integral_le_of_norm_le_const {a b C : ℝ} {f : ℝ → E}
+  (h : ∀ x ∈ Ioc (min a b) (max a b), ∥f x∥ ≤ C) :
+  ∥∫ x in a..b, f x∥ ≤ C * abs (b - a) :=
+norm_integral_le_of_norm_le_const_ae $ eventually_of_forall h
+
+lemma integral_add (hfm : measurable f) (hfi : interval_integrable f μ a b)
+  (hgm : measurable g) (hgi : interval_integrable g μ a b) :
+  ∫ x in a..b, f x + g x ∂μ = ∫ x in a..b, f x ∂μ + ∫ x in a..b, g x ∂μ :=
+begin
+  simp only [interval_integral, integral_add hfm hfi.1 hgm hgi.1,
+    integral_add hfm hfi.2 hgm hgi.2],
+  abel
+end
+
+@[simp] lemma integral_neg : ∫ x in a..b, -f x ∂μ = -∫ x in a..b, f x ∂μ :=
+begin
+  simp only [interval_integral, integral_neg],
+  abel
+end
+
+variables [topological_space α] [opens_measurable_space α]
+
+section order_closed_topology
+
+variables [order_closed_topology α]
+
+lemma integral_cocycle (hfm : measurable f) (hab : interval_integrable f μ a b)
```

Comment:
I don't really like this name: to me, the cocyle property is more like the next lemma. Something like `integral_add_adjacent_intervals_cancel` is longer, but more evocative to me.

Reply:
Done.